src/common/pico_sync/include/pico/mutex.h

```diff
@@ -28,19 +28,33 @@ extern "C" {
  *
  * See \ref critical_section.h for protecting access between multiple cores AND IRQ handlers
  */
-
 typedef struct __packed_aligned mutex {
     lock_core_t core;
-    int8_t owner; //! core number or -1 for unowned
+    lock_owner_id_t owner;      //! owner id LOCK_INVALID_OWNER_ID for unowned
+    uint8_t recursion_state;    //! 0 means non recursive (owner or unowned)
```

Comment:
I wonder if it'd have been worth making this `uint8_t` a typedef (e.g. `recursion_state_t`) ? :shrug:   Or perhaps we already have enough typedefs scattered throughout the place.